### DISCUSSION TITLE
[crazyswarm_server-2] process has died

### DISCUSSION THREAD

**Discussion by TheLastPixie** [USC-ACTLab/crazyswarm#711] [Q&A]
Hi everyone, 
I'm wondering if I can get some help with this issue I have seen this brought up twice before but the scenarios were different.

-  in our case, we just got crazyflie 2.1 this week and it works fine from the cfclient.

- The problem is ROS related, we followed the installation steps exactly as shown crazyswarm repository, I have installed, and uninstalled ROS and crazyswarm multiple times but to no avail. (made sure to uninstall properly every time)
- the crazyradio connects normally via the `rosrun crazyflie_tools scan -v`
- also one more thing the chooser.py always gives me an error when we run a battery test. dunno if that is going to help
- when we edit the yaml file to disable motion capture we get the following error:
`[crazyswarm_server-2] process has died [pid 76390, exit code -6, cmd /home/user/crazyswarm/ros_ws/devel/lib/crazyswarm/crazyswarm_server __name:=crazyswarm_server __log:=/home/user/.ros/log/0b313522-6603-11ed-b787-49934f717d49/crazyswarm_server-2.log].
log file: /home/user/.ros/log/0b313522-6603-11ed-b787-49934f717d49/crazyswarm_server-2*.log`
Ubuntu version: 20.04
ROS: Noetic

Any kind of help would be appreciated

**Comment by knmcguire** [comment]
Hi! Could you provide that particular log?

It could also be that you changed the Yaml file in a way that it can not be read out properly anymore, could you share the full content of the yaml file too?

**Reply by TheLastPixie** [reply]
[crazyswarmlog.zip](https://github.com/USC-ACTLab/crazyswarm/files/10077980/crazyswarmlog.zip)
[yaml files.zip](https://github.com/USC-ACTLab/crazyswarm/files/10077981/yaml.files.zip)

Thank you alot for reaching out, find the crazyswarm yaml and log files.

**Reply by whoenig** [reply]
I did a quick glance and didn't spot a particular issue.

**Comment by knmcguire** [comment]
Also, there is not a lot of maintaince happening on crazyswarm based on ROS1, so if you are able to use ROS2, then perhaps it would be better to look there: https://github.com/IMRCLab/crazyswarm2

**Reply by TheLastPixie** [reply]
We tried building crazyswarm2 in ros2, ros2 installs and builds normally, but crazyswarm2 doesn't build correctly no matter what we do. I will try my hand at it today again.

**Reply by whoenig** [reply]
You can open a discussion in the crazyswarm2 repo. As with Crazyswarm1, there is continuous integration on github, so building should work.

**Reply by TheLastPixie** [reply]
Hi Dr. @whoenig , thank you for your replies I will build crazyswarm2 on a different machine and see what I get, thank you for your feedback, I guess I would have to dig in the C++ code to figure out the server error, my question would be what does the error message indicate if that might help.

**Reply by whoenig** [reply]
The error in Crazyswarm1 indicates a segmentation fault. Most of the time, this means that the configuration file contains something that we didn't anticipate/check for to provide a better error message.
